apprentice_teacher.rb

```diff
@@ -1,21 +1,12 @@
-class ApprenticeTeacher
-  attr_reader :age, :salary, :phase, :target_raise
-  attr_accessor :name
+require_relative 'faculty'
+
+class ApprenticeTeacher < Faculty
 
   def initialize(options={})
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
-    @target_raise = 800
+    super
     @phase = 3
```

Comment:
Looks great!  The one remaining bit of repeated code that I can see is that you set `@phase = 3` in both ApprenticeTeacher and Teacher.  I'd pull that out into Faculty, or even into SchoolMember, but overriding it after calling super in Student.